apollo-ios/Sources/ApolloWebSocket/WebSocketTransport.swift

```diff
@@ -450,49 +455,52 @@ extension WebSocketTransport: NetworkTransport {
     operation: Operation,
     cachePolicy: CachePolicy,
     contextIdentifier: UUID? = nil,
-    context: (any RequestContext)? = nil,
-    callbackQueue: DispatchQueue = .main,
-    completionHandler: @escaping @Sendable (Result<GraphQLResult<Operation.Data>, any Error>) -> Void) -> any Cancellable {
-
-      @Sendable func callCompletion(with result: Result<GraphQLResult<Operation.Data>, any Error>) {
-        callbackQueue.async {
-          completionHandler(result)
-        }
-      }
-
-      if let error = self.error {
-        callCompletion(with: .failure(error))
-        return EmptyCancellable()
+    context: (any RequestContext)? = nil
+  ) throws -> AsyncThrowingStream<GraphQLResult<Operation.Data>, any Error> {
+    if let error = self.error {
+      return AsyncThrowingStream.init {
+        throw error
       }
+    }
 
-      return WebSocketTask(self, operation) { [weak store, contextIdentifier] result in
-        Task {
-          do {
-            let jsonBody = try result.get()
-            let response = GraphQLResponse(operation: operation, body: jsonBody)
+    #warning("TODO: stream never finishes. WebSocketTask does not report subscription termination.")
```

Comment:
I think it has been a request though and now that we have async streams we can signal when that stream is 'done' vs. just never calling a closure again.